test/functional/adb-commands-e2e-specs.js

```diff
@@ -98,4 +99,34 @@ describe('adb commands', function () {
     logs.should.have.length.above(0);
     await adb.stopLogcat();
   });
+  describe('app permissions', async () => {
+    before(async function () {
+      let deviceApiLevel = await adb.getApiLevel();
+      if (deviceApiLevel < 23) {
+        //test should skip if the device API < 23
+        this.skip();
+      }
+      let isInstalled = await adb.isAppInstalled('io.appium.android.apis');
+      if (isInstalled) {
+        await adb.uninstallApk('io.appium.android.apis');
+      }
+    });
+    it('should install and grant all permission', async () => {
+      let apiDemos = path.resolve(rootDir, 'test',
+          'fixtures', 'ApiDemos-debug.apk');
+      await adb.install(apiDemos);
+      (await adb.isAppInstalled('io.appium.android.apis')).should.be.true;
+      let requestedPermissions = await adb.getReqPermissions('io.appium.android.apis');
+      expect(await adb.getReqPermissions('io.appium.android.apis')).to.have.members(requestedPermissions);
```

Comment:
I'm not sure if this is testing anything. What it seems to be doing is ensuring that calling `getReqPermissions` twice returns the same thing. Is that what you want to be testing?